### PR TITLE
OADP-3598: Update logic to wait for pull secret in pod restore plugin

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
This PR updates the logic to check whether OpenShift Image Registry is enabled or not.

### CODE REVIEW

**Comment by shubham-pampattiwar** [comment]
Testing on OCP 4.14.z with registry enabled:

```shubham@fedora ~/oadp-operator -  (master) $ oc logs velero-7cb9f7bc7f-8xd2t | grep shubham
Defaulted container "velero" out of: velero, openshift-velero-plugin (init), velero-plugin-for-aws (init), velero-plugin-for-csi (init)
time="2024-03-15T16:12:52Z" level=info msg="shubham imageRegistryEnabled: true" cmd=/plugins/velero-plugins logSource="/opt/app-root/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/pod/restore.go:333" pluginName=velero-plugins restore=openshift-adp/testnsbackup-1-20240315091156
time="2024-03-15T16:12:52Z" level=info msg="shubham majorVersionInt: 4, minorVersionInt: 14" cmd=/plugins/velero-plugins logSource="/opt/app-root/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/pod/restore.go:340" pluginName=velero-plugins restore=openshift-adp/testnsbackup-1-20240315091156
time="2024-03-15T16:12:52Z" level=info msg="shubham WaitForPullSecrets: true" cmd=/plugins/velero-plugins logSource="/opt/app-root/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/pod/restore.go:215" pluginName=velero-plugins restore=openshift-adp/testnsbackup-1-20240315091156
shubham@fedora ~/oadp-operator -  (master) $ oc version
Client Version: 4.14.7
Kustomize Version: v5.0.1
Server Version: 4.14.11
Kubernetes Version: v1.27.10+28ed2d7
```

Testing on OCP 4.15.z with registry disabled:

```
shubham@fedora ~/oadp-operator -  (master) $ oc logs velero-85c64d7f8c-wnkx6 | grep shubham
Defaulted container "velero" out of: velero, openshift-velero-plugin (init), velero-plugin-for-aws (init), velero-plugin-for-csi (init)
time="2024-03-15T16:15:56Z" level=info msg="shubham imageRegistryEnabled: false" cmd=/plugins/velero-plugins logSource="/opt/app-root/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/pod/restore.go:333" pluginName=velero-plugins restore=openshift-adp/testnsbackup-1-20240315091555
time="2024-03-15T16:15:56Z" level=info msg="shubham majorVersionInt: 4, minorVersionInt: 15" cmd=/plugins/velero-plugins logSource="/opt/app-root/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/pod/restore.go:340" pluginName=velero-plugins restore=openshift-adp/testnsbackup-1-20240315091555
time="2024-03-15T16:15:56Z" level=info msg="shubham WaitForPullSecrets: false" cmd=/plugins/velero-plugins logSource="/opt/app-root/src/github.com/konveyor/openshift-velero-plugin/velero-plugins/pod/restore.go:215" pluginName=velero-plugins restore=openshift-adp/testnsbackup-1-20240315091555
shubham@fedora ~/oadp-operator -  (master) $ oc version
Client Version: 4.14.7
Kustomize Version: v5.0.1
Server Version: 4.15.0
Kubernetes Version: v1.28.6+6216ea1
shubham@fedora ~/oadp-operator -  (master) $ 
```

**Review by mateusoliveira43 — DISMISSED** [review]

File: velero-plugins/util/openshift/imageregistry.go [hunk 1]
```diff
@@ -35,26 +30,3 @@ func ImageRegistryCapabilityEnabled() (bool, error) {
...
-		return false, err
-	}
-	return c.Status.ReadyReplicas > 0, nil
-}
-
-// https://github.com/openshift/cluster-image-registry-operator/blob/48875d3ccb4595be9d3bec563d1fda2eb940cecf/pkg/defaults/defaults.go#L19
-// avoiding indirect imports and version conflicts
-const ImageRegistryResourceName = "cluster"
-
-func GetImageRegistryConfig() (*v1.Config, error) {
```
Comment (kaovilai) [inline velero-plugins/util/openshift/imageregistry.go]: I think we still need this

If capabilities is enabled we [need another check](https://redhat.enterprise.slack.com/archives/C013VBYBJQH/p1710168733543119?thread_ts=1709952439.633439&cid=C013VBYBJQH) on the registry config.

> it's the managementState (.spec.managementState: Removed) @ Tiger, not replicas. Let me see if I can find some docs for reference

Comment (shubham-pampattiwar) [reply velero-plugins/util/openshift/imageregistry.go]: Updated the PR

**Review by sseago — APPROVED** [review]

**Review by weshayutin — APPROVED** [review]
/LGTM

**Pull request merged**